src/main/java/org/veupathdb/service/eda/ds/plugin/standalonemap/CollectionFloatingBarplotPlugin.java

```diff
@@ -67,32 +67,37 @@ protected void validateVisualizationSpec(CollectionFloatingBarplotSpec pluginSpe
   }
 
   @Override
-  protected List<StreamSpec> getRequestedStreams(CollectionFloatingBarplotSpec pluginSpec) {    
+  protected List<StreamSpec> getRequestedStreams(CollectionFloatingBarplotSpec pluginSpec) {
+    String outputEntityId = pluginSpec.getOutputEntityId();
+    List<VariableSpec> plotVariableSpecs = new ArrayList<VariableSpec>();
+    plotVariableSpecs.addAll(pluginSpec.getOverlayConfig().getSelectedMembers());
+
     return ListBuilder.asList(
-      new StreamSpec(DEFAULT_SINGLE_STREAM_NAME, pluginSpec.getOutputEntityId())
-        .addVars(pluginSpec.getOverlayConfig().getSelectedMembers())
-      ); 
+      new StreamSpec(DEFAULT_SINGLE_STREAM_NAME, outputEntityId)
+        .addVars(plotVariableSpecs)
+        // TODO can we make this automagical?
```

Comment:
Not clear to me what you want to be automagical?  We probably can with some inheritance if you want to generalize some things.  Just hit me up on slack if you want to discuss.  Looks like there might be a lot of repetition/cut-and-paste in some of these plugins which we would like to avoid.

Reply:
there is a lot of repetition, and i hate it, and every time i think about improving the situation it seems to turn into a can of worms.. i start thinking things like 'well in an ideal world id actually be doing some entirely other thing' and things spiral out of control. so if you have ideas, id be open to hearing them